lib/v2/fda/cdrh.js

```diff
@@ -0,0 +1,60 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const rootUrl = 'https://www.fda.gov';
+    const currentUrl = `${rootUrl}/medical-devices/news-events-medical-devices/cdrhnew-news-and-updates`;
+
+    const response = await got({
+        method: 'get',
+        url: currentUrl,
+    });
+
+    const $ = cheerio.load(response.data);
+
+    let items = $('div[role="main"] a')
+        .slice(0, ctx.query.limit ? parseInt(ctx.query.limit) : 30)
+        .toArray()
+        .map((item) => {
+            item = $(item);
+
+            const link = item.attr('href');
+
+            return {
+                title: item.text(),
+                link: `${/^http/.test(link) ? '' : String(rootUrl)}${link}`,
+            };
+        });
+
+    items = await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: item.link,
+                });
+
+                const content = cheerio.load(detailResponse.data);
+
+                item.author = content('meta[property="article:publisher"]').attr('content');
+
+                try {
+                    item.pubDate = parseDate(content('meta[property="article:modified_time"]').attr('content').split(', ').pop(), 'MM/DD/YYYY - HH:mm');
```

Comment:
Use `meta[property="article:published_time"]` instead?
Some articles like https://www.federalregister.gov/d/2022-13954 do not have `modified_time` and using `parseDate()` will return the current time instead.

Reply:
You are right ☺️